src/bridges/tokemak/DepositBridge.sol

```diff
@@ -0,0 +1,337 @@
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import { IDefiBridge } from "../../interfaces/IDefiBridge.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IRollupProcessor} from '../../interfaces/IRollupProcessor.sol';
+
+import { AztecTypes } from "../../aztec/AztecTypes.sol";
+interface Ttoken is IERC20{
+  function requestedWithdrawals(address account) external view returns (uint256, uint256);
+  function requestWithdrawal(uint256 amount) external;
+  function deposit(uint256 amount) external payable;
+  function withdraw(uint256 requestedAmount) external;
+  function withdraw(uint256 requestedAmount, bool asEth) external;
+}
+interface IManager {
+  function getCurrentCycleIndex() external view returns (uint256);
+}
+contract DepositBridge is IDefiBridge {
+  using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+    event TokenBalance(uint256 balance);
+
+  address public tWETH = 0xD3D13a578a53685B4ac36A1Bab31912D2B2A2F36;
+  address public tUSDC  = 0x04bDA0CF6Ad025948Af830E75228ED420b0e860d;
+  address public tDAI  = 0x0CE34F4c26bA69158BC2eB8Bf513221e44FDfB75;
+  address public tFRAX  = 0x94671A3ceE8C7A12Ea72602978D1Bb84E920eFB2;
+  address public tWUST  = 0x482258099De8De2d0bda84215864800EA7e6B03D;
+
+  address public WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+  address public USDC = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+  address public DAI = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+  address public FRAX = 0x853d955aCEf822Db058eb8505911ED77F175b99e;
+  address public wUST = 0xa693B19d2931d498c5B318dF961919BB4aee87a5;
+
+  address public MANAGER = 0xA86e412109f77c45a3BC1c5870b880492Fb86A14;
+
+  address public immutable rollupProcessor;
+
+
+
+  uint256 internal constant MAX_UINT = type(uint256).max;
+
+  uint256 internal constant MIN_GAS_FOR_CHECK_AND_FINALISE = 83000;
+  uint256 internal constant MIN_GAS_FOR_FUNCTION_COMPLETION = 5000;
+
+
+  mapping(address=> uint256) pendingWithdarawls;
+
+  struct Interaction {
+      uint256 inputValue;
+      address tAsset;
+  }
+
+
+  uint256[] nonces;
+  // cache of all of our Defi interactions. keyed on nonce
+  mapping(uint256 => Interaction) public pendingInteractions;
+
+  constructor(address _rollupProcessor) public {
+    rollupProcessor = _rollupProcessor;
+    
+  }
+
+  function convert(
+    AztecTypes.AztecAsset memory inputAssetA,
+    AztecTypes.AztecAsset memory inputAssetB,
+    AztecTypes.AztecAsset memory outputAssetA,
+    AztecTypes.AztecAsset memory outputAssetB,
+    uint256 totalInputValue,
+    uint256 interactionNonce,
+    uint64 auxData,
+    address rollupBeneficiary
+  )
+    external
+    payable
+    override
+    returns (
+      uint256 outputValueA,
+      uint256 outputValueB,
+      bool isAsync
+    )
+  {
+    // // ### INITIALIZATION AND SANITY CHECKS
+    require(msg.sender == rollupProcessor, "Tokemak DepositBridge: INVALID_CALLER");
```

Comment:
I think we should sanity check all of the following inputs:

- inputAssetA == ERC20 (as you have done
- inputAssetB == NOT_USED
- outputAssetA == ERC20
- outputAssetB == NOT_USED
- is it possible to check that the outputAssetA address is the correct one for the given input asset?
- totalInputValue > 0